java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -0,0 +1,519 @@
+                        break;
+                    case Types.DOUBLE:
+                        updateVector((Float8Vector)root.getVector(columnName),
+                                rs.getDouble(i), !rs.wasNull(), rowCount);
+                        break;
+                    case Types.CHAR:
+                    case Types.NCHAR:
+                    case Types.VARCHAR:
+                    case Types.NVARCHAR:
+                    case Types.LONGVARCHAR:
+                    case Types.LONGNVARCHAR:
+                        updateVector((VarCharVector)root.getVector(columnName),
+                                rs.getString(i), !rs.wasNull(), rowCount);
+                        break;
+                    case Types.DATE:
+                        updateVector((DateMilliVector) root.getVector(columnName),
+                                rs.getDate(i, calendar), !rs.wasNull(), rowCount);
+                        break;
+                    case Types.TIME:
+                        updateVector((TimeMilliVector) root.getVector(columnName),
+                                rs.getTime(i, calendar), !rs.wasNull(), rowCount);
+                        break;
+                    case Types.TIMESTAMP:
+                        // TODO: Need to handle precision such as milli, micro, nano
+                        updateVector((TimeStampVector)root.getVector(columnName),
+                                rs.getTimestamp(i, calendar), !rs.wasNull(), rowCount);
+                        break;
+                    case Types.BINARY:
+                    case Types.VARBINARY:
+                    case Types.LONGVARBINARY:
+                        updateVector((VarBinaryVector)root.getVector(columnName),
+//                                rs.getBytes(i), !rs.wasNull(), rowCount);
+                                rs.getBinaryStream(i), !rs.wasNull(), rowCount);
+                        break;
+                    case Types.ARRAY:
+                        // TODO Need to handle this type
+                    	// fields.add(new Field("list", FieldType.nullable(new ArrowType.List()), null));
+                        break;
+                    case Types.CLOB:
+                        updateVector((VarCharVector)root.getVector(columnName),
+                            rs.getClob(i), !rs.wasNull(), rowCount);
+                        break;
+                    case Types.BLOB:
+                        updateVector((VarBinaryVector)root.getVector(columnName),
+                            rs.getBlob(i), !rs.wasNull(), rowCount);
+                        break;
+
+                    default:
+                        // no-op, shouldn't get here
+                        break;
+                }
+            }
+            rowCount++;
+        }
+        root.setRowCount(rowCount);
+    }
+
+    private static void updateVector(BitVector bitVector, boolean value, boolean isNonNull, int rowCount) {
+        NullableBitHolder holder = new NullableBitHolder();
```

Comment:
is it better to use the holder vs calling directly `bitVector.setSafe(rowCount, isNonNull ? 1 : 0, value ? 1: 0)` (cc @siddharthteotia )

Reply:
I think we can continue to use holder so as to be consistent with other parts of the code. What do you think?

Reply:
I think its fine to use any APIs exposed by vectors